File: asc/semis_reports/report/special_students_summary_with_level/special_students_summary_with_level.js
// Copyright (c) 2016, Frappe Technologies and contributors
// For license information, please see license.txt
/* eslint-disable */

let loc_options = ["", "Urban", "Rural"];
let status_options = ["", "Functional", "Closed"];
let gender_options = ["", "Boys", "Girls", "Mixed"];
frappe.query_reports["Special Students Summary With Level"] = {
  filters: [
    {
      fieldname: "year",
      label: __("Year"),
      fieldtype: "Link",
      options: "Year",
      default: "2021-22",
			reqd: 1
    },
    {
      fieldname: "division",
      label: __("Division"),
      fieldtype: "Link",
      options: "Division",
    },
{
      fieldname: "district",
      label: __("District"),
      fieldtype: "Link",
      options: "District",
      "get_query": function() {
			var division = frappe.query_report.get_filter_value('division');
			if (division) {
			  return {
				"doctype": "District",
				"filters": {
				  "division": division,
				}
			  }
			}
		  }
    },
    {
      fieldname: "location",
      label: __("Location"),
      fieldtype: "Select",
      options: loc_options,
    },
    {
      fieldname: "status",
      label: __("Status"),
      fieldtype: "Select",
      options: status_options,
    },
    {
      fieldname: "gender",
      label: __("Gender"),
      fieldtype: "Select",
      options: gender_options,
    },
  ],
};

File: asc/semis_reports/report/special_students_summary_with_level/special_students_summary_with_level.py
# Copyright (c) 2013, Frappe Technologies and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
	columns, data = [], []
	columns = get_columns()
	data = get_data(filters)
	return columns, data

def get_columns():
	columns = [
		{
		"label": _("Level"),
		"fieldtype": "Link",
		"fieldname": "level",
		"options": "Level",
		"width": 150
	},
	{
		"label": _("Total Disables"),
		"fieldtype": "Int",
		"fieldname": "total_disables",
		"width": 150
	},
	{
		"label": _("Physical Disables"),
		"fieldtype": "Int",
		"fieldname": "physical_disables",
		"width": 150
	},
	{
		"label": _("Polio Affected"),
		"fieldtype": "Int",
		"fieldname": "polio_affected",
		"width": 150
	},
	{
		"label": _("Wheel Chair Ramps"),
		"fieldtype": "Int",
		"fieldname": "wheel_chair_ramps",
		"width": 150
	},
	]
	return columns

def get_data(filters):
	conditions, group_by = get_condition(filters)

	temp_query = """SELECT tabLevel.name , 
 	SUM(physical_disabilites + polio_affected), 
  	SUM(physical_disabilites),	SUM(polio_affected), 
   	SUM(CASE WHEN wheel_chair_ramp_available = 'Yes' Then 1 Else 0 END) FROM tabASC CROSS JOIN tabLevel ON tabASC.level = tabLevel.name  
    WHERE tabASC.docstatus != 2 %s %s ORDER BY tabLevel.list_order asc""" % (
			conditions, group_by)

	info = frappe.db.sql(temp_query, filters)

	return info

def get_condition(filters):
	conditions , group_by = "" , "GROUP BY tabLevel.name"

	if filters.get("division"):
		conditions += " AND region = %(division)s"
	if filters.get("year"):
		conditions += "  AND  year = %(year)s"
	if filters.get("location"):
		conditions += "  AND  location = %(location)s"
	if filters.get("status"):
		conditions += "  AND  status_detail = %(status)s"
	if filters.get("district"):
		conditions += " AND district = %(district)s"
	if filters.get("gender"):
		conditions += " AND school_gender = %(gender)s"

	return conditions, group_by
    